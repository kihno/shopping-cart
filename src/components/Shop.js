import React from 'react';
import Product from './Product';

const Shop = (props) => {

    return (
        <div id="shop">
            {props.items.map(item => {
                return <Product item={item} />
            })}
        </div>
    )
}

export default Shop;