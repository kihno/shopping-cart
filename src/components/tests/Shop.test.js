import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Shop from '../Shop';

jest.mock('../Product', () => (props) => {
    return (<div data-testid="item" key={props.item.key}>{props.item.name}</div>);
});

const items = [
    {name: 'foo', key: 0},
    {name: 'bar', key: 1},
    {name: 'baz', key: 2},
];

describe('Shop page', () => {
    it ('renders shop page', () => {
        const { container } = render(<Shop items={items} />);
        expect(container).toMatchSnapshot();
    });

    it ('renders product listings', () => {
        render(<Shop items={items} />);

        expect(screen.queryAllByTestId('item').length).toBe(3);
        expect(screen.queryAllByTestId('item')[0].textContent).toBe('foo');
    });
})