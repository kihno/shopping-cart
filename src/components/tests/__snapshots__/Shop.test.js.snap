// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop page renders shop page 1`] = `
<div>
  <div
    id="shop"
  >
    <div
      data-testid="item"
    >
      foo
    </div>
    <div
      data-testid="item"
    >
      bar
    </div>
    <div
      data-testid="item"
    >
      baz
    </div>
  </div>
</div>
`;
